.github/workflows/tests.yml

```diff
@@ -0,0 +1,185 @@
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        toolchain: [nightly, beta, stable, 1.34]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: ${{ matrix.toolchain }}
+          override: true
+      - run: cargo test
+      - run: cargo test --features std
+      - if: ${{ matrix.toolchain == 'nightly' }}
+        run: cargo build --benches
+
+  linux-tests:
+    name: Additional Linux targets
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        target: [
+          x86_64-unknown-linux-musl,
+          i686-unknown-linux-gnu,
+          i686-unknown-linux-musl,
+        ]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          target: ${{ matrix.target }}
+          toolchain: stable
+          override: true
+      - run: sudo apt install gcc-multilib
+      - run: cargo test --target ${{ matrix.target }}
+
+  windows-tests:
+    name: Additional Windows targets
+    runs-on: windows-latest
+    strategy:
+      matrix:
+        toolchain: [
+          stable-x86_64-gnu,
+          stable-i686-gnu,
+          stable-i686-msvc,
+        ]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: ${{ matrix.toolchain }}
+          override: true
+      - run: cargo test
+
+  cross-tests:
+    name: Cross tests
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        include:
+          - os: ubuntu-latest
+            target: mips-unknown-linux-gnu
+            toolchain: stable
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          target: ${{ matrix.target }}
+          toolchain: ${{ matrix.toolchain }}
+          override: true
+      - name: Cache cargo plugins
+        uses: actions/cache@v1
+        with:
+          path: ~/.cargo/bin/
+          key: ${{ runner.os }}-cargo-plugins
+      - name: Install cross
+        run: cargo install cross || true
+      - name: Test
+        run: cross test --no-fail-fast --target ${{ matrix.target }}
+
+  build:
+    name: Build-only
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        target: [
+          x86_64-sun-solaris,
+          x86_64-unknown-freebsd,
+          x86_64-fuchsia,
+          x86_64-unknown-netbsd,
+          x86_64-unknown-redox,
+          x86_64-fortanix-unknown-sgx,
+        ]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          target: ${{ matrix.target }}
+          toolchain: nightly
```

Comment:
Should we be using the stable toolchain for these targets?

Reply:
Travis config was using nightly, so I haven't changed it. I can try `stable` instead.

Reply:
On a second thought, I think it's worth to keep it `nightly` to allow early compiler bug detection if there will be any affecting us. I also will enable the cron job option to run CI periodically.